lib/common/output_html.c

```diff
@@ -417,6 +418,10 @@ pcmk__output_create_html_node(pcmk__output_t *out, const char *element_name, con
 
     CRM_ASSERT(out != NULL);
 
+    if (!pcmk__str_eq(out->fmt_name, "html", pcmk__str_none)) {
+        return NULL;
+    }
+
```

Comment:
I think if functions aren't using out->priv, we don't need to check. If someone finds a twisted use for one of these functions, it's no big deal unless it would break :)

Reply:
Some of these functions return a value, which means I also need to make sure the callers of those functions are okay with getting a NULL.

Reply:
Having worked on this a little bit, I feel like it probably makes the most sense to CRM_ASSERT() instead of checking for NULL and returning.  I'm not sure we can do anything useful if one of these functions returns NULL.